spec/index.bs

```diff
@@ -14488,6 +14534,21 @@ An <dfn dfn>Origin3D</dfn> is a {{GPUOrigin3D}}.
         either {{GPUOrigin3DDict}}.{{GPUOrigin3DDict/z}}
         or the third item of the sequence or 0 if it isn't present.
 </div>
+<div algorithm>
+    <dfn abstract-op>validate GPUOrigin3D shape</dfn>(origin)
+
+    **Arguments:**
+
+    - |origin|: The {{GPUOrigin3D}} to validate.
+
+   **Returns:** {{undefined}}
+
+    1. Throw a {{TypeError}} if:
+
+       - |origin| is a sequence, and
+       - |origin|.length &lt; 0 or |origin|.length &gt; > 3.
```

Comment:
Same as above. Can probably be:
`Throw a {{TypeError}} if |origin| is a sequence and |origin|.length &gt; 3.`